src/components/modals/ModalUserEdit.tsx

```diff
@@ -0,0 +1,363 @@
+    setPathError("");
+    if (firstTimeSetup && !path) {
+      setPathError(t("modalscandirectoryedit.mustspecifypath"));
+      return;
+    }
+    if (path) {
+      const result = !findPath(treeData, path);
+      setPathDoesNotExist(result);
+      if (result) {
+        setPathError(t("modalscandirectoryedit.pathdoesnotexist"));
+      }
+    }
+  };
+
+  const clearStateAndClose = () => {
+    setUserName("");
+    setUserEmail("");
+    setUserFirst("");
+    setUserLast("");
+    setUserPassword("");
+    setNewScanDirectory("");
+
+    setUserNameError("");
+    setUserEmailError("");
+    setClosing(false);
+    onRequestClose();
+  };
+
+  const validateAndClose = () => {
+    setClosing(true);
+
+    validateUsername(userName);
+    validateEmail(userEmail);
+    validatePath(newScanDirectory);
+
+    var newUserData = { ...userToEdit };
+
+    if (newScanDirectory) {
+      newUserData.scan_directory = newScanDirectory;
+    }
+    if (!newUserData.scan_directory) {
+      delete newUserData.scan_directory;
+    }
+
+    if (userNameError || !newPasswordIsValid || userEmailError || pathError) {
+      return;
+    }
+    if (createNew) {
+      if (userPassword && userName) {
+        signup(userName, userPassword, userEmail, userFirst, userLast, false, dispatch, true, newScanDirectory);
+        clearStateAndClose();
+      }
+      return;
+    } else {
+      newUserData.email = userEmail;
+      newUserData.first_name = userFirst;
+      newUserData.last_name = userLast;
+
+      if (userPassword) {
+        newUserData.password = userPassword;
+      }
+      if (userName) {
+        newUserData.username = userName;
+      }
+    }
+
+    if (updateAndScan) {
+      dispatch(updateUserAndScan(newUserData));
+    } else {
+      dispatch(manageUpdateUser(newUserData));
+    }
+    clearStateAndClose();
+  };
+  const onPasswordValidate = (pass: string, valid: boolean) => {
+    setUserPassword(pass);
+    setNewPasswordIsValid(valid);
+  };
+
+  return (
+    <Modal
+      opened={isOpen}
+      centered
+      overflow="outside"
+      size="xl"
+      onClose={() => {
+        clearStateAndClose();
+      }}
+      title={<Title order={4}>{createNew ? t("modaluseredit.createheader") : t("modaluseredit.header")}</Title>}
+    >
+      <Box
+        sx={theme => ({
+          paddingBottom: theme.spacing.md,
+        })}
+      >
+        <SimpleGrid cols={2} spacing="xs" breakpoints={[{ maxWidth: 600, cols: 1, spacing: "sm" }]}>
+          <TextInput
+            required
+            label={t("login.usernamelabel")}
+            icon={<User />}
+            placeholder={t("login.usernameplaceholder")}
+            name="username"
+            value={userName}
+            error={userNameError}
+            onChange={event => {
+              userToEdit.username = event.currentTarget.value;
+              setUserName(userToEdit.username);
+              validateUsername(event.currentTarget.value);
```

Comment:
inline functions like these get recreated on every render. In this case every time the user types a single letter.  This also applies to the functions below. Would making use of the `useCallback` hook make sense here?